examples/control_arm/forward_analysis.py

```diff
@@ -0,0 +1,263 @@
+            (BOUNDS_X[0] < x)
+            & (x < BOUNDS_X[1])
+            & (BOUNDS_Y[0] < y)
+            & (y < BOUNDS_Y[1])
+            & (BOUNDS_Z[0] < z)
+            & (z < BOUNDS_Z[1])
+        ),
+        weight_dict={
+            "equilibrium_x": "sdf",
+            "equilibrium_y": "sdf",
+            "equilibrium_z": "sdf",
+            "stress_disp_xx": "sdf",
+            "stress_disp_yy": "sdf",
+            "stress_disp_zz": "sdf",
+            "stress_disp_xy": "sdf",
+            "stress_disp_xz": "sdf",
+            "stress_disp_yz": "sdf",
+        },
+        name="INTERIOR",
+    )
+
+    # re-assign to cfg.TRAIN.iters_per_epoch
+    if cfg.TRAIN.enable_parallel:
+        cfg.TRAIN.iters_per_epoch = len(arm_left_constraint.data_loader)
+
+    # wrap constraints togetherg
+    constraint = {
+        arm_left_constraint.name: arm_left_constraint,
+        arm_right_constraint.name: arm_right_constraint,
+        arm_surface_constraint.name: arm_surface_constraint,
+        arm_interior_constraint.name: arm_interior_constraint,
+    }
+
+    # initialize solver
+    solver = ppsci.solver.Solver(
+        model_list,
+        constraint,
+        cfg.output_dir,
+        optimizer,
+        lr_scheduler,
+        cfg.TRAIN.epochs,
+        cfg.TRAIN.iters_per_epoch,
+        save_freq=cfg.TRAIN.save_freq,
+        log_freq=cfg.log_freq,
+        eval_during_train=cfg.TRAIN.eval_during_train,
+        eval_freq=cfg.TRAIN.eval_freq,
+        seed=cfg.seed,
+        equation=equation,
+        geom=geom,
+        checkpoint_path=cfg.TRAIN.checkpoint_path,
+    )
+
+    # train model
+    solver.train()
+
+    # plot losses
+    solver.plot_loss_history(by_epoch=True, smooth_step=1)
+
+
+def evaluate(cfg: DictConfig):
+    # set random seed for reproducibility
+    ppsci.utils.misc.set_random_seed(cfg.seed)
+    # initialize logger
+    logger.init_logger("ppsci", osp.join(cfg.output_dir, f"{cfg.mode}.log"), "info")
+
+    # set model
+    disp_net = ppsci.arch.MLP(**cfg.MODEL.disp_net)
+    stress_net = ppsci.arch.MLP(**cfg.MODEL.stress_net)
+    # wrap to a model_list
+    model_list = ppsci.arch.ModelList((disp_net, stress_net))
+
+    # set geometry
+    control_arm = ppsci.geometry.Mesh(cfg.GEOM_PATH)
+    # geometry bool operation
+    geo = control_arm
+    geom = {"geo": geo}
+    # set bounds
+    BOUNDS_X, BOUNDS_Y, BOUNDS_Z = control_arm.bounds
+
+    # set visualizer(optional)
+    # add inferencer data
+    samples = geom["geo"].sample_interior(
+        cfg.TRAIN.batch_size.visualizer_vtu,
+        criteria=lambda x, y, z: (
+            (BOUNDS_X[0] < x)
+            & (x < BOUNDS_X[1])
+            & (BOUNDS_Y[0] < y)
+            & (y < BOUNDS_Y[1])
+            & (BOUNDS_Z[0] < z)
+            & (z < BOUNDS_Z[1])
+        ),
+    )
+    pred_input_dict = {}
+    for key in samples:
+        if key in cfg.MODEL.disp_net.input_keys:
+            pred_input_dict[key] = samples[key]
```

Comment:
`pred_input_dict = {k: v for k, v in samples.items() if k in cfg.MODEL.disp_net.input_keys}`